pkg/controller/master/virtualmachine/datavolume_util.go

```diff
@@ -0,0 +1,41 @@
+package virtualmachine
+
+import (
+	kubevirtapis "kubevirt.io/client-go/api/v1alpha3"
+	cdiapis "kubevirt.io/containerized-data-importer/pkg/apis/core/v1beta1"
+
+	cdictrl "github.com/rancher/harvester/pkg/generated/controllers/cdi.kubevirt.io/v1beta1"
+	"github.com/rancher/harvester/pkg/ref"
+)
+
+// adoptDataVolumeIfOrphan tries to refer the target VirtualMachine as the owner of the DataVolume.
+func adoptDataVolumeIfOrphan(dataVolumeClient cdictrl.DataVolumeClient, dv *cdiapis.DataVolume, vm *kubevirtapis.VirtualMachine) error {
+	if vm == nil || dv == nil || dv.DeletionTimestamp != nil {
+		return nil
+	}
+	var dvHandler, orphan, err = ref.RecordOwner(dv, vm, kubevirtapis.VirtualMachineGroupVersionKind)
+	if err != nil {
+		return err
+	}
+	if !orphan {
+		return nil
+	}
+	_, err = dataVolumeClient.Update(dvHandler.(*cdiapis.DataVolume))
+	return err
+}
+
+// abandonDataVolumeIfOwned tries to un-refer the target VirtualMachine as the owner of the DataVolume.
```

Comment:
un-refer is not a word, change to `unset the dataVolume's owner reference of its target VM`.

Reply:
Updated